contracts/core/mpl-rewards/v1/test/helper/Rewards.sol

```diff
@@ -0,0 +1,740 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.11;
+pragma experimental ABIEncoderV2;
+
+import "test/TestUtil.sol";
+import "test/user/Custodian.sol";
+import "./IStakeToken.sol";
+
+contract CustodialTestHelper is TestUtil {
+
+    using SafeMath for uint256;
+
+    function setupFarmingEcosystem() internal {
+        setUpGlobals();
+        setUpTokens();
+        setUpOracles();
+        setUpFactories();
+        setUpCalcs();
+        setUpActors();
+        createBalancerPool(100_000 * USD, 10_000 * USD);
+        transferBptsToPoolDelegatesAndStakers();
+        setUpLiquidityPool();
+        setUpMplRewardsFactory();
+    }
+
+    function custody_and_transfer(uint256 amt, uint256 custodyAmt1, uint256 custodyAmt2, bool isPfdtStakeAsset, IStakeToken stakeToken) public {
+        Custodian custodian1 = new Custodian();  // Custodial contract for FDTs - will start out as liquidity mining but could be broader DeFi eventually
+        Custodian custodian2 = new Custodian();  // Custodial contract for FDTs - will start out as liquidity mining but could be broader DeFi eventually
+
+        if (isPfdtStakeAsset) {
+            amt         = constrictToRange(amt,         100, 1E9,     true);  // $1 - $1b
```

Comment:
We should make this much bigger. `1e9` is `1e-9 WAD`, we should do something like
`amt         = constrictToRange(amt,         100, stakeToken.balanceOf(address(sam),     true);  // $1 - $1b`